core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -229,18 +252,34 @@ protected ManifestEntry<DataFile> prepare(ManifestEntry<DataFile> entry) {
     @Override
     protected FileAppender<ManifestEntry<DataFile>> newAppender(
         PartitionSpec spec, OutputFile file) {
+      return newAppender(spec, file, null, null);
+    }
+
+    @Override
+    protected FileAppender<ManifestEntry<DataFile>> newAppender(
+        PartitionSpec spec, OutputFile file, String compressionCodec, Integer compressionLevel) {
       Schema manifestSchema = V2Metadata.entrySchema(spec.partitionType());
       try {
-        return Avro.write(file)
-            .schema(manifestSchema)
-            .named("manifest_entry")
-            .meta("schema", SchemaParser.toJson(spec.schema()))
-            .meta("partition-spec", PartitionSpecParser.toJsonFields(spec))
-            .meta("partition-spec-id", String.valueOf(spec.specId()))
-            .meta("format-version", "2")
-            .meta("content", "data")
-            .overwrite()
-            .build();
+        Avro.WriteBuilder builder =
+            Avro.write(file)
+                .schema(manifestSchema)
+                .named("manifest_entry")
+                .meta("schema", SchemaParser.toJson(spec.schema()))
+                .meta("partition-spec", PartitionSpecParser.toJsonFields(spec))
+                .meta("partition-spec-id", String.valueOf(spec.specId()))
+                .meta("format-version", "2")
+                .meta("content", "data")
+                .overwrite();
+
+        if (compressionCodec != null) {
+          builder.set(TableProperties.AVRO_COMPRESSION, compressionCodec);
+        }
+
+        if (compressionLevel != null) {
+          builder.set(TableProperties.AVRO_COMPRESSION_LEVEL, compressionLevel.toString());
```

Comment:
Why not pass a string argument?

Reply:
In the original PR (https://github.com/apache/iceberg/pull/5893), @sumeetgajjar used a String for compressionLevel  in the API initially, and @rdblue commented that compression level is an int, not a string. Sumeet then changed it to an Integer, to allow for the fact that some codecs do not have compression level (so it can be null).
I think that conceptually, it makes sense for compressionLevel to be an Integer in the API. In the implementation, it so happens that we call `Avro.WriteBuilder` and that makes us use a `set(String, String)` method to configure it, and ironically `Avro.WriteBuilder.Context` has to convert the String for compression level back into an int. But that is all implementation detail and an accident of the implementation.

Reply:
On the other hand, I agree that making compressionLevel a String is more efficient due to how the implementation works. One can make an argument for either case, it just depends on what you prioritize.
I don't have a strong opinion on this. For now, I'll keep this as it is.

Reply:
Thanks for the background.